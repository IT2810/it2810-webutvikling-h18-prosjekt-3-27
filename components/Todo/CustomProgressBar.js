import {Text, View} from "react-native";
import React, {Component} from "react";
import ProgressBar from 'react-native-progress/Bar';
export default class CustomProgressBar extends Component {

  render() {
    return(
      <View style={{flexDirection: "row", alignItems: "center", justifyContent: "center"}}>
        <Text style={{marginRight: 5}}>{this.props.numCompleted}</Text>
         <ProgressBar progress={this.getProgress()} color="#579d5b"/>
        <Text style={{marginLeft: 5}}>{this.props.numUncompleted}</Text>
      </View>
    )
  }

  getProgress() {
    return isNaN(this.props.progress) ? 0 : this.props.progress;
  }
}
