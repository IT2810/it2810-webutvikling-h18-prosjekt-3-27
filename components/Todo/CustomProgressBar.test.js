import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import CustomProgressBar from "./CustomProgressBar";


test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(<CustomProgressBar numCompleted={1}
                                                  numUncompleted={1}
                                                  progress={1} />);
  expect(tree).toMatchSnapshot();
});