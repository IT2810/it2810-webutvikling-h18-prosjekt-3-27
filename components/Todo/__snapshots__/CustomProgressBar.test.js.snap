// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginRight": 5,
      }
    }
  >
    1
  </Text>
  <ProgressBar
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    borderRadius={4}
    borderWidth={1}
    color="#579d5b"
    height={6}
    indeterminate={false}
    progress={1}
    useNativeDriver={false}
    width={150}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginLeft": 5,
      }
    }
  >
    1
  </Text>
</Component>
`;
